lake-primitives/src/types/receipts.rs

```diff
@@ -254,7 +252,76 @@ impl From<&views::ActionView> for Operation {
             views::ActionView::Delegate {
                 delegate_action,
                 signature,
-            } => todo!(),
+            } => Self::Delegate {
+                delegate_action: delegate_action.into(),
+                signature: signature.clone(),
+            },
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct DelegateAction {
+    pub sender_id: AccountId,
+    pub receiver_id: AccountId,
+    /// List of actions to be executed.
+    ///
+    /// With the meta transactions MVP defined in NEP-366, nested
+    /// DelegateActions are not allowed. A separate type is used to enforce it.
+    pub operations: Vec<NonDelegateOperation>,
+    pub nonce: u64,
+    /// The maximal height of the block in the blockchain below which the given DelegateAction is valid.
+    pub max_block_height: u64,
+    pub public_key: PublicKey,
+}
+
+impl From<&near_primitives::delegate_action::DelegateAction> for DelegateAction {
+    fn from(delegate_action: &near_primitives::delegate_action::DelegateAction) -> Self {
+        let operations: Vec<NonDelegateOperation> = delegate_action
+            .actions
+            .iter()
+            .cloned()
+            .map(Into::into)
+            .collect();
+
+        Self {
+            sender_id: delegate_action.sender_id.clone(),
+            receiver_id: delegate_action.receiver_id.clone(),
+            operations,
+            nonce: delegate_action.nonce,
+            max_block_height: delegate_action.max_block_height,
+            public_key: delegate_action.public_key.clone(),
+        }
+    }
+}
+
+impl From<near_primitives::delegate_action::NonDelegateAction> for NonDelegateOperation {
+    fn from(non_delegate_action: near_primitives::delegate_action::NonDelegateAction) -> Self {
+        non_delegate_action.into()
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct NonDelegateOperation(Operation);
```

Comment:
Can't say I am excited about this implementation, though I've replicated it from the `nearcore` itself.

My initial thought was to go through the check like `NonDelegateAction` and after that to put `Operation` directly to the `DelegateAction.operation` but I guess we have to use the `NonDelegateOperation` wrapper there for the sanity.